internal/common/pointers.go

```diff
@@ -3,3 +3,10 @@ package common
 func ToPtr[T any](x T) *T {
 	return &x
 }
+
+func FromPtr[T any](ref *T) (value T) {
```

Comment:
(not for this PR but I want to raise awareness as my idea is not really getting traction): a maybe interesting approach for this could be taken from https://github.com/osbuild/images/pull/860 - i.e. instead of wrapping pointers (with the risk of still dereferencing them and causing panics) we could move into a world of optional datatypes that encapsulate the pointers entirely so that a panic is never possible. Anyway, this totally fine in this context of course, just trying to raise awareness :)

Reply:
How rustful :)